### ISSUE TITLE
setup module returns completely different info for windows vs linux

### ISSUE THREAD

**Issue by kefoster** [ansible/ansible#15506]
##### ISSUE TYPE

- Bug Report
##### ANSIBLE VERSION

```
ansible 2.0.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides

```
##### SUMMARY

I wanted to pull the memory for all my hosts using the setup module.  I ran the following:
$ sudo ansible all -m setup -a 'filter=ansible_memtotal_mb' --ask-vault-pass
I was expecting to just get the memory for all my systems but windows returns the memory different:
Linux reports like this:
0.10.60.0 | SUCCESS => {
    "ansible_facts": {
        "ansible_memtotal_mb": 3823
    },
    "changed": false
}

Windows reports like this:
redacted | SUCCESS => {
    "ansible_facts": {
        "ansible_architecture": "64-bit",
        "ansible_date_time": {
            "date": "4/20/2016",
            "day": "20",
            "hour": "13",
            "iso8601": "2016-04-20T13:45:44",
            "minute": "45",
            "month": "04",
            "year": "2016"
        },
        "ansible_distribution": "Microsoft Windows NT 6.3.9600.0",
        "ansible_distribution_version": "6.3.9600.0",
        "ansible_fqdn": "redacted.com",
        "ansible_hostname": "redacted",
        "ansible_interfaces": [
            {
                "default_gateway": "re.dac.te.d",
                "dns_domain": "redacted.com",
                "interface_index": 14,
                "interface_name": "vmxnet3 Ethernet Adapter"
            }
        ],
        "ansible_ip_addresses": [
            "re.dac.te.d"
        ],
        "ansible_lastboot": "2016-03-30 09:18:11Z",
        "ansible_os_family": "Windows",
        "ansible_os_name": "Microsoft Windows Server 2012 R2 Standard",
        "ansible_powershell_version": 4,
        "ansible_system": "Win32NT",
        "ansible_totalmem": 4294496256,
        "ansible_uptime_seconds": 1830453,
        "ansible_winrm_certificate_expires": "2017-03-29 09:54:35"
    },
    "changed": false
}

I would expect that the setup module would return common info memory, cpu, memory.. the same.  
##### STEPS TO REPRODUCE

Get a combined windows and linux system environment and run the following:
sudo ansible all -m setup -a 'filter=ansible_memtotal_mb' --ask-vault-pass

```
<!--- Paste example playbooks or commands between quotes -->
```

##### EXPECTED RESULTS

0.10.60.0 | SUCCESS => {
    "ansible_facts": {
        "ansible_memtotal_mb": 3823
    },
    "changed": false
}

redacted | SUCCESS => {
    "ansible_facts": {
        "ansible_memtotal_mb": 3823
    },
    "changed": false
}
##### ACTUAL RESULTS

```
<!--- Paste verbatim command output between quotes -->
```

**Comment by jhawkesworth** [comment]
Setup against windows hosts could definitely use expanded facts, although there will always be differences.

Would be nice to support filter against windows hosts too.

**Comment by jhawkesworth** [comment]
A lot of extra properties were merged under #3777 with common names.

Is there something specific you need changing or can this now be close?

needs_info

**Comment by dagwieers** [comment]
Yes, the reported issue was fixed with ansible/ansible-modules-core#3777

I think this issue can be closed. If anything specific is still missing or incomplete, open a new issue or send a PR.

**Comment by dagwieers** [comment]
@gundalow @jhawkesworth Please close.
